add delay option

jonhadfield opened the pull request:
Adds an option `-d` to specify the number of milliseconds to pause between each request, per task.  
My own use-case is to reduce load on a site I'm crawling for cache warming, but also useful to prevent app/WAF rate-limiting.

XORbit01 commented:
Thanks for the pull request!

you suggested and implemented a good feature!

but I am afraid to say this will affect the parallel traversing when you don't specify `--live` option
so when you did sleep, each concurrent task will sleep however another concurrent task will start, it will make usefulness  time interval, because in this case the program is traversing not sequentially.


I suggest to:
specify the delay only in live mode, so you can move the logic (sleep) `CrawlNodeLive`

and later we can implement the delay for the parallel traversal ie `block mode`

jonhadfield commented:
Hi @XORbit01, I've moved delay to live mode only and modified the cli options slightly.  

wrt. a delay in block mode, I now understand. Whilst I think there's benefit to introducing a delay when acting concurrently, I doubt there'll be much call for such a feature. IMHO, concurrency with "think time" makes more sense in a performance testing tool.

XORbit01 commented:
good work! 
it will be merged asap
